upgrade/upgrade-scripts/execute

```diff
@@ -28,6 +30,24 @@ function usage() {
 	exit 2
 }
 
+function generate_interface_to_mac_address_map() {
+	# we always create the map file anew to avoid using any stale versions of the same
+	if [[ -e "$INTERFACE_TO_MACADRESS_MAP_FILE_PATH" ]]; then
+		rm "$INTERFACE_TO_MACADRESS_MAP_FILE_PATH" ||
+			die "Could not delete the existing contents of the mac address map file"
+	fi
+	for i in $(find /sys/class/net/ -type l | awk -F "/" '{print $5}'); do
+		interface_name="$i"
+		if [ -e /sys/class/net/"$i"/address ] &&
+			grep -q -v -e "00:00:00:00:00:00" /sys/class/net/"$i"/address; then
```

Comment:
I don't think you need to worry about excluding the address for the `lo` interface given that this one won't appear in the netplan file. It'll be a harmless entry in your map file, and simplifies this code.

Reply:
Sure.